src/hooks/useSampleData.js

```diff
@@ -0,0 +1,28 @@
+import { useState, useEffect } from 'react';
+import { getSampleUserData, getSampleFolderData } from '../api/api';
+
+const useSampleData = dataType => {
+  const [data, setData] = useState(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        let sampleData;
+        if (dataType === 'user') {
+          sampleData = await getSampleUserData();
+        } else if (dataType === 'folder') {
+          sampleData = await getSampleFolderData();
+        }
+        setData(sampleData);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    fetchData();
+  }, [dataType]);
+
+  return data;
+};
```

Comment:
잘 구현해 주셨는데, dataType이 추가될 때마다 else if를 추가해야 한다는 불편함이 있겠네요!
이 경우, 객체로 매핑하면 좀 더 깔끔하게 구현할 수 있습니다.
대략적으로 다음과 같이 구현해볼 수 있겠네요.

```suggestion
const SAMPLE_DATA_MAP = {
  user: getSampleUserData,
  foler: getSampleFolderData,
};

const useSampleData = dataType => {
  const [data, setData] = useState(null);

  useEffect(() => {
    const fetchData = async () => {
      try {
        const sampleData = SAMPLE_DATA_MAP[dataType]?.();
        // 중간에 optional chaining(?.)은 위 객체에서 매핑할 수 없는 경우에 대한 에러 방지용입니다.
        
        setData(sampleData);
      } catch (error) {
        console.error(error);
      }
    };

    fetchData();
  }, [dataType]);

  return data;
};
```